src/views/Checkout.vue

```diff
@@ -18,13 +18,58 @@
 <script lang="ts">
 import { Component, Emit, Vue } from 'vue-property-decorator';
 import { PaymentInfoLine, SmallPage } from '@nimiq/vue-components';
-import { ParsedCheckoutRequest } from '../lib/RequestTypes';
-import { Static } from '../lib/StaticStore';
+import { ParsedCheckoutRequest, RequestType } from '../lib/RequestTypes';
+import { State as RpcState } from '@nimiq/rpc';
+import staticStore, { Static } from '../lib/StaticStore';
+import { LoginResult } from '@/lib/RequestTypes';
+import { WalletStore } from '@/lib/WalletStore';
+import { WalletInfo } from '../lib/WalletInfo';
+import { Mutation } from 'vuex-class';
 
 @Component({components: {PaymentInfoLine, SmallPage}})
 export default class Checkout extends Vue {
+    @Static private rpcState!: RpcState;
     @Static private request!: ParsedCheckoutRequest;
 
+    @Mutation('addWallet') private $addWallet!: (walletInfo: WalletInfo) => any;
+    @Mutation('setActiveAccount') private $setActiveAccount!: (payload: {
+        walletId: string,
+        userFriendlyAddress: string,
+    }) => any;
+
+    private addWallet() {
+        // Redirect to import
+        const request: KeyguardRequest.ImportRequest = {
+            appName: this.request.appName,
+            defaultKeyPath: `m/44'/242'/0'/0'`,
+            requestedKeyPaths: [`m/44'/242'/0'/0'`],
+        };
+
+        staticStore.originalRequestType = RequestType.CHECKOUT;
```

Comment:
As this is used exclusively as a route later I suggest renaming it that and store the actual route name 
```suggestion
        staticStore.originalRouteName = RequestType.CHECKOUT;
```
There is potential for the originalRoute not being RequestName, but maybe further into the request.